pegasusio/zarr_utils.py

```diff
@@ -303,10 +313,16 @@ def write_dataframe(self, parent: zarr.Group, name: str, df: pd.DataFrame) -> No
         attrs_dict['index_name'] = df.index.name if df.index.name is not None else 'index'
         self.write_series(group, '_index', df.index.values)
         for col in df.columns:
-            self.write_series(group, col, df[col].values)
+            if isinstance(df[col].values[0], np.ndarray):
+                colgroup = group.create_group(col, overwrite = True)
+                x = 0
+                for data in df[col].values:
+                    x = x+1
+                    self.write_series(colgroup, col + str(x), data )
```

Comment:
Accordingly, convert `data` to numpy array here, i.e. only when writing to zarr files.